src/ipx.ts

```diff
@@ -196,8 +190,44 @@ class IPX {
     operations.forEach(({ operation, args }) => {
       sharp = operation.handler(context, sharp, ...args) || sharp
     })
+    return sharp
+  }
+
+  async getData (info: IPXImageInfo) {
+    // Check cache existence
+    const cache = await this.cache!.get(info.cacheKey)
+    if (cache) {
+      return cache
+    }
+    const sharp = await this.applyOperations(info)
     const data = await sharp.toBuffer()
```

Comment:
We can check here if `info.format` endsWith `.json`, wrap data into JSON container (so no more duplicate logic for `getJSONData`. Cache key is already unique as format is different

Reply:
We can't keep `.json` in format because later we are checking valid formats and use this format as a `sharp` operation.
But I've extracted the logic from middleware and move it to info function

Reply:
We can split by `.` and pick first partition for ipx format :) (this way we can support `.brotli` too later on)